src/ripple/core/impl/Config.cpp

```diff
@@ -379,253 +382,253 @@ void Config::load ()
+    if (ifsConfig.bad ())
+    {
+        std::cerr << "Failed to read '" << CONFIG_FILE << "'." << std::endl;
+        return;
+    }
 
-            if (smtTmp)
-            {
-                CLUSTER_NODES = *smtTmp;
-            }
+    loadFromString (fileContents);
+}
 
-            smtTmp  = getIniFileSection (secConfig, SECTION_IPS);
+void Config::loadFromString (std::string const& fileContents)
+{
+    IniFileSections secConfig = parseIniFile (fileContents, true);
 
-            if (smtTmp)
-            {
-                IPS = *smtTmp;
-            }
+    build (secConfig);
 
-            smtTmp  = getIniFileSection (secConfig, SECTION_IPS_FIXED);
+    if (auto smtTmp =
+        getIniFileSection (secConfig, SECTION_VALIDATORS))
+    {
+        validators  = *smtTmp;
+    }
 
-            if (smtTmp)
-            {
-                IPS_FIXED = *smtTmp;
-            }
+    if (auto smtTmp =
+        getIniFileSection (secConfig, SECTION_CLUSTER_NODES))
+    {
+        CLUSTER_NODES = *smtTmp;
+    }
 
-            smtTmp = getIniFileSection (secConfig, SECTION_SNTP);
+    if (auto smtTmp =
+        getIniFileSection (secConfig, SECTION_IPS))
+    {
+        IPS = *smtTmp;
+    }
 
-            if (smtTmp)
-            {
-                SNTP_SERVERS = *smtTmp;
-            }
+    if (auto smtTmp =
+        getIniFileSection (secConfig, SECTION_IPS_FIXED))
+    {
+        IPS_FIXED = *smtTmp;
+    }
 
-            smtTmp  = getIniFileSection (secConfig, SECTION_RPC_STARTUP);
+    if (auto smtTmp =
+        getIniFileSection (secConfig, SECTION_SNTP))
+    {
+        SNTP_SERVERS = *smtTmp;
+    }
 
-            if (smtTmp)
-            {
-                RPC_STARTUP = Json::arrayValue;
+    if (auto smtTmp =
+        getIniFileSection (secConfig, SECTION_RPC_STARTUP))
+    {
+        RPC_STARTUP = Json::arrayValue;
 
-                for (auto const& strJson : *smtTmp)
-                {
-                    Json::Reader    jrReader;
-                    Json::Value     jvCommand;
+        for (auto const& strJson : *smtTmp)
+        {
+            Json::Reader    jrReader;
+            Json::Value     jvCommand;
 
-                    if (!jrReader.parse (strJson, jvCommand))
-                        throw std::runtime_error (
-                            boost::str (boost::format (
-                                "Couldn't parse [" SECTION_RPC_STARTUP "] command: %s") % strJson));
+            if (!jrReader.parse (strJson, jvCommand))
+                throw std::runtime_error (
+                    boost::str (boost::format (
+                        "Couldn't parse [" SECTION_RPC_STARTUP "] command: %s") % strJson));
 
-                    RPC_STARTUP.append (jvCommand);
-                }
-            }
+            RPC_STARTUP.append (jvCommand);
+        }
+    }
 
-            if (getSingleSection (secConfig, SECTION_DATABASE_PATH, DATABASE_PATH))
-                DATA_DIR    = DATABASE_PATH;
+    {
```

Comment:
Doesn't `BasicConfig::legacy` perform essentially the same function as `getSingleSection` except that it returns the value instead of setting a non-const reference parameter?